tests/kernel/mem_protect/syscalls/src/main.c

```diff
@@ -377,6 +405,28 @@ void test_syscall_context(void)
 
 K_MEM_POOL_DEFINE(test_pool, BUF_SIZE, BUF_SIZE, 4 * NR_THREADS, 4);
 
+/**
+ * @brief Test to demonstrate system call is handled on a separate stack
+ *
+ * @details - Created a syscall with name test_syscall_separate_stack()
+ * - Inside of that syscall was created a local variable,
+ * and then using thread->stack_info was taken start address
+ * and size of stack of the caller thread
+ * - Then inside of the system call address of the local variable
+ * checked that it is not within the bounds of the caller thread stack
+ * - If that test is running in user mode, verify that
+ * system call is handled on a private stack,
```

Comment:
not sure what "private" means here, I guess the explanation in the last line is sufficient. Maybe replace private with "separate"